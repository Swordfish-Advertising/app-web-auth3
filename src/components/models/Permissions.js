// @flow

type Permission = {
  streamId: string,
  level: 'read'|'contribute'|'manage',
  defaultName: ?string,
};

export type PermissionsList = Array<Permission>;

class Permissions {
  list: PermissionsList;

  constructor (permissionsList: string|PermissionsList) {
    if (typeof permissionsList === 'string') {
      this.list = JSON.parse(permissionsList);
    } else {
      this.list = permissionsList;
    }
  }

  updateList (newList: PermissionsList): PermissionsList {
    this.list = newList;
    return this.list;
  }
}

export default Permissions;
