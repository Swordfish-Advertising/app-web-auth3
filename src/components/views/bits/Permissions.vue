<template>
  <v-form>
    <h1>Permissions</h1>

    <div id="appIdText">
      App <b>{{ appId }}</b> is requesting:
    </div>

    <ul>
      <li
        v-for="(permission, index) in permissionsList"
        :key="index">
        A permission on stream <b>{{ permission.streamId }}</b> with level <b>{{ permission.level }}</b>
      </li>
    </ul>

    <div v-if="clientData.consentMsg">
      {{ clientData.consentMsg }}
    </div>

    <v-btn
      id="refusePermissions"
      @click="()=>this.$emit('refused')"
    >Reject</v-btn>

    <v-btn
      id="acceptPermissions"
      @click="()=>this.$emit('accepted')"
    >Accept</v-btn>

  </v-form>
</template>

<script>
export default {
  props: {
    permissionsList: {type: Array, default: () => ([])},
    appId: {type: String, default: ''},
    clientData: {type: Object, default: () => ({})},
  },
};
</script>
