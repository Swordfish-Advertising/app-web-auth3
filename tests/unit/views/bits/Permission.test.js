import { shallowMount } from '@vue/test-utils';
import Permissions from '@/components/views/bits/Permissions';
import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

describe('Permissions.test.js', () => {
  let wrapper;
  const appId = 'testApp';
  const permissions = [
    {streamId: 'diary', level: 'read'},
    {streamId: 'work', level: 'manage'},
  ];

  beforeAll(() => {
    wrapper = shallowMount(Permissions, {
      propsData: {
        permissionsArray: permissions,
        appId: appId,
        accept: () => {},
        refuse: () => {},
      },
    });
  });

  it('renders correctly (snapshots matching)', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('shows the id of the requesting app', () => {
    const appIdText = wrapper.find('#appIdText');
    expect(appIdText.html()).toContain(`App <b>${appId}</b> is requesting:`);
  });

  it('shows a list of all requested permissions', () => {
    const list = wrapper.find('ul').html();
    permissions.forEach(permission => {
      expect(list).toContain(`A permission on stream <b>${permission.streamId}</b> with level <b>${permission.level}</b></li>`);
    });
  });
});
