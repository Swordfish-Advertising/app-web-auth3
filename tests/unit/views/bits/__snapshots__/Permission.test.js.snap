// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permissions.test.js renders correctly (snapshots matching) 1`] = `
<vform-stub>
  <h1>
    Permissions
  </h1>
   
  <div
    id="appIdText"
  >
    
    App 
    <b>
      testApp
    </b>
     is requesting:
  
  </div>
   
  <ul>
    <li>
      
      A permission on stream 
      <b>
        diary
      </b>
       with level 
      <b>
        read
      </b>
    </li>
    <li>
      
      A permission on stream 
      <b>
        work
      </b>
       with level 
      <b>
        manage
      </b>
    </li>
  </ul>
   
  <!---->
   
  <vbtn-stub
    activeclass="v-btn--active"
    id="refusePermissions"
    ripple="true"
    tag="button"
    type="button"
  >
    Reject
  </vbtn-stub>
   
  <vbtn-stub
    activeclass="v-btn--active"
    id="acceptPermissions"
    ripple="true"
    tag="button"
    type="button"
  >
    Accept
  </vbtn-stub>
</vform-stub>
`;
